components/AddAgreement/AddAgreement.js

```diff
@@ -1,56 +1,90 @@
 import { useRouter } from 'next/router'
 import { useAgreementContract } from "../../contexts/ContractContext"
+import { useAccount } from "../../contexts/AccountContext"
 import { useState } from "react"
 import styles from "./AddAgreement.module.css"
 import Image from 'next/image'
 import { BeatLoader } from "react-spinners"
 import { ethers } from "ethers"
-const BigNumber = ethers.BigNumber
+import * as Yup from 'yup';
 
 function AddAgreementForm(props) {
   const [title, setTitle] = useState("")
   const [description, setDescription] = useState("")
-  const [developer, setDeveloper] = useState("")
+  const [professional, setProfessional] = useState("")
   const [skills, setSkills] = useState("")
   const [paymentAmount, setPaymentAmount] = useState("")
   const [isLoading, setIsLoading] = useState(false)
   const [transactionHash, setTransactionHash] = useState(null)
   const { contract, loading } = useAgreementContract()
+  const [formErrors, setFormErrors] = useState({});
   const router = useRouter()
+  const { account } = useAccount();
+
+  const AgreementSchema = Yup.object().shape({
+    title: Yup.string()
+      .required('Title is required'),
```

Comment:
these texts should not be hard coded but rather be in a translation dictionary so the platform can be easily translated to other languages